Classes/ViewHelpers/SimplePrevNextViewHelper.php

```diff
@@ -168,20 +173,36 @@ protected function getNeighbours(News $news, $pidList, $sortField)
         $data = [];
         $pidList = empty($pidList) ? $news->getPid() : $pidList;
 
-        $connection = GeneralUtility::makeInstance(ConnectionPool::class)
-            ->getConnectionForTable('tx_news_domain_model_news');
-
         foreach (['prev', 'next'] as $label) {
-            $queryBuilder = $connection->createQueryBuilder();
-
-            $extraWhere = [];
-            if ((bool)$this->arguments['includeInternalType'] === false) {
-                $extraWhere[] = $queryBuilder->expr()->neq('type', $queryBuilder->createNamedParameter(1, \PDO::PARAM_INT));
-            }
-            if ((bool)$this->arguments['includeExternalType'] === false) {
-                $extraWhere[] = $queryBuilder->expr()->neq('type', $queryBuilder->createNamedParameter(2, \PDO::PARAM_INT));
+            $row = $this->getNeighbour($label, $pidList, $sortField, $news);
+            if (is_array($row)) {
+                $data[$label] = $row;
+            } else {
```

Comment:
this can be combined as elseif with next if

Reply:
Thanks!